### PR TITLE
feat: do not add constraints at branches when forking is disabled && …

### PR DESCRIPTION
**Pull request by yype**
feat: do not add constraints at branches when forking is disabled && add support for setting this on or off

### CODE REVIEW

**Review by Changochen — COMMENTED** [review]
Looks good. But as we discussed, we will not merge it now.

File: klee/include/klee/ExecutionState.h [hunk 1]
```diff
@@ -78,6 +78,9 @@ class ExecutionState : public IAddressSpaceNotification {
     /// Disables forking, set by user code.
     bool forkDisabled;
 
+    /// Whether to add the constraints at branches when forking is disabled
+    bool addConstraintWhenForkingIsDisabled;
```
Comment (Changochen) [inline klee/include/klee/ExecutionState.h]: Or you can mark this as true by default, then we can merge it as it is enabled by default.

Comment (yype) [reply klee/include/klee/ExecutionState.h]: I've made it an argument that defaults to true.

**Review by Changochen — APPROVED** [review]
LGTM

**Pull request merged**